lib/byron/src/Cardano/Wallet/Byron/Network.hs

```diff
@@ -617,6 +626,10 @@ instance ToText NetworkLayerLog where
             [ "Error when querying local state parameters:"
             , T.pack e
             ]
+        MsgDestroyCursor threadId -> T.unwords
+            [ "Destroying cursor connection at"
```

Comment:
Was there a specific story for the addition of `destroyCursor`? It was called when workers crashed, right? You noticed chain sync threads staying around?

Reply:
Yeah, this is a bit of a side-story within the story. While I was debugging the worker looping and crashing I noticed that we weren't actually closing connections to the node and eventually, the system would run out of file descriptors. 

These doesn't quite fix the problem though as the it seems that even killing the thread doesn't close the connection (not immediately at least) and I was still able to reach the maximum number of sockets allowed by the system fairly quickly :upside_down_face: ... that's a first step in the right direction I believe. Perhaps some little delay before looping in case of errors would be nice to give time for the connection to be cleared up.